File: AdhomPanel/AdhomPanel.Web/Modules/AdhomStore/category/categoryForm.cs
using Serenity.ComponentModel;
using Serenity.Web;

namespace AdhomPanel.AdhomStore.Forms
{
    [FormScript("AdhomStore.category")]
    [BasedOnRow(typeof(categoryRow), CheckNames = true)]
    public class categoryForm
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public string Image { get; set; }
    }
}

File: AdhomPanel/AdhomPanel.Web/Modules/ServerTypes/AdhomStore/categoryForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface categoryForm {
    Name: StringEditor;
    Description: StringEditor;
    Image: StringEditor;
}

export class categoryForm extends PrefixedContext {
    static formKey = 'AdhomStore.category';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!categoryForm.init)  {
            categoryForm.init = true;

            var w0 = StringEditor;

            initFormType(categoryForm, [
                'Name', w0,
                'Description', w0,
                'Image', w0
            ]);
        }
    }
}
